q2_sample_classifier/tests/test_classifier.py

```diff
@@ -84,31 +84,114 @@ def _load_md(md_fp):
+        regress_samples(tmpd, self.table_ecam_fp, self.mdc_ecam_fp,
+                        test_size=0.5, cv=3,
+                        n_estimators=2, n_jobs=-1,
+                        estimator='RandomForestRegressor')
+
+    # test that each regressor works and delivers an expected accuracy result
+    # when a random seed is set.
+    def test_regressors(self):
+        for regressor in ['RandomForestRegressor', 'ExtraTreesRegressor',
+                          'GradientBoostingRegressor', 'AdaBoostRegressor',
+                          'Lasso', 'Ridge', 'ElasticNet',
+                          'KNeighborsRegressor', 'LinearSVR', 'SVR']:
             tmpd = join(self.temp_dir.name, regressor)
             mkdir(tmpd)
-            regress_samples(tmpd, self.table_ecam_fp, self.md_ecam_fp,
-                            category='month', test_size=0.5, cv=3,
-                            n_estimators=2, n_jobs=-1, estimator=regressor)
-
+            estimator, pd, pt = _set_parameters_and_estimator(
+                regressor, self.table_ecam_fp, self.md_ecam_fp, 'month',
+                n_estimators=10, n_jobs=-1, cv=1,
+                random_state=123, parameter_tuning=False, classification=False)
+            estimator, cm, accuracy, importances = split_optimize_classify(
+                self.table_ecam_fp, self.md_ecam_fp, 'month', estimator,
+                tmpd, test_size=0.5, cv=1, random_state=123,
+                n_jobs=-1, optimize_feature_selection=False,
+                parameter_tuning=False, param_dist=None, classification=False,
+                calc_feature_importance=False, scoring=mean_squared_error)
+            self.assertAlmostEqual(accuracy, seeded_results[regressor])
+
+    # test some invalid inputs/edge cases
+    def test_invalids(self):
+        estimator, pd, pt = _set_parameters_and_estimator(
+            'RandomForestClassifier', self.table_chard_fp, self.md_chard_fp,
+            'Region', n_estimators=10, n_jobs=-1, cv=1,
+            random_state=123, parameter_tuning=False, classification=True)
+        regressor, pd, pt = _set_parameters_and_estimator(
+            'RandomForestRegressor', self.table_chard_fp, self.md_chard_fp,
+            'Region', n_estimators=10, n_jobs=-1, cv=1,
+            random_state=123, parameter_tuning=False, classification=True)
+        with self.assertRaises(ValueError):
+            # zero samples (if mapping file and table have no common samples)
+            estimator, cm, accuracy, importances = split_optimize_classify(
+                self.table_ecam_fp, self.md_chard_fp, 'Region', estimator,
+                self.temp_dir.name, test_size=0.5, cv=1, random_state=123,
+                n_jobs=-1, optimize_feature_selection=False,
+                parameter_tuning=False, param_dist=None,
+                calc_feature_importance=False)
+            # too few samples to stratify
+            estimator, cm, accuracy, importances = split_optimize_classify(
```

Comment:
You need to include `with self.assertRaises(ValueError):` before each of these tests. If not, the test will pass as long as one of these three commands raises a ValueError. 

Also, we generally recommend using `assertRaisesRegex`, which tests whether the error message that is raised matches some pattern. This is useful for making sure that you are getting a `ValueError` for the reason that you expect. See [here](https://github.com/qiime2/q2-diversity/blob/93aa393308750cbfae0716c5df89114412741253/q2_diversity/tests/test_alpha.py#L96) and [here](https://github.com/qiime2/q2-diversity/blob/93aa393308750cbfae0716c5df89114412741253/q2_diversity/tests/test_alpha.py#L100) for a good example of this.